### PR TITLE
Add predicate StakePoolCostTooLowPOOL

### LINKED ISSUE #909: Ledger predicate: StakePoolCostTooLowPOOL
### Abstract

Implement predicate `StakePoolCostTooLowPOOL`

### Why?

[Conformance](https://github.com/IntersectMBO/cardano-ledger/blob/0cfbf861cfb456660a7b73281c6fb714a53d40f9/eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Pool.hs#L253-L262)

### How?

[Reference Implementation](https://github.com/IntersectMBO/cardano-ledger/blob/0cfbf861cfb456660a7b73281c6fb714a53d40f9/eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Pool.hs#L253-L262)

### Testing Strategy / Acceptance Criteria

Introduce fixtures for predicate coverage

### Discussion points

_No response_

### Dependencies & Related Tasks

_No response_

### Checklist

- [x] I understand that feature requests and unrefined work item should be open as [GitHub Discussions](https://github.com/pragma-org/amaru/discussions) instead.
- [x] I have assigned this item to an existing milestone from the roadmap
- [x] I have added a label capturing the impact of this item (i.e. value for users/stakeholders if successful)
- [x] I have added a label capturing the delivery risk of this item (i.e. how likely is it that this task will succeed as planned)
- [x] I have added a label capturing the effort of this item (i.e. how large is the task?)

### PR DESCRIPTION
**Pull request by jonathanlim222**
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Phase One stake pool validation to report “pool cost too low” failures using the correct predicate, including the boundary where the cost falls below the minimum.
* **Tests**
  * Added new Phase One failure fixtures for stake pool cost-too-low scenarios (including the minimum-boundary case) to strengthen regression coverage.
* **Documentation**
  * Updated the unreleased changelog entry and reference links for the new coverage.

### CODE REVIEW

**Review by yHSJ — APPROVED** [review]

**Review by KtorZ — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ Other guiding principles:
 ### Added
 
 - **amaru-ledger**: reject treasury withdrawal proposals that reference unregistered reward accounts.  ([#1032][], [#929][])
-
+- **amaru-ledger**: introduce `StakePoolCostTooLowPOOL` coverage. ([#1037][], [#909][])
```
Comment (KtorZ) [inline CHANGELOG.md:44]: Note: I try to avoid mentioning internal jargon in the changelog; it's supposed to be a user-facing piece of information; and no one knows what `StakePoolCostTooLowPOOL` is.
